lib/resty/http.lua

```diff
@@ -174,8 +174,23 @@ function _M.ssl_handshake(self, ...)
     return sock:sslhandshake(...)
 end
 
+do
+    local new_connect = require "resty.http_connect"
+    -- Old function signatures to support:
+    -- ok, err = httpc:connect(host, port, options_table?)
+    -- ok, err = httpc:connect("unix:/path/to/unix.sock", options_table?)
+    function _M.connect(self, options, ...)
+        if type(options) == "table" then
+            -- new all-in-one interface
+            return new_connect(self, options)
+        else
+            -- backward compatible
+            return self:old_connect(options, ...)
+        end
+    end
+end
 
-function _M.connect(self, ...)
+function _M.old_connect(self, ...)
     local sock = self.sock
     if not sock then
         return nil, "not initialized"
```

Comment:
I prefer the way this is explained in the docs to the way to code reads - i.e. "all-in-one" vs "tcp-only". Can we avoid naming the original function `old_connect`? To me I think it makes sense to have `connect` dispatch to either `tcp_only_connect` or `all_in_one_connect`, based on the arg types.